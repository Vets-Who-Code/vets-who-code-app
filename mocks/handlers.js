import { rest } from 'msw'

/**
 * The handlers description
 * @export handlers[] list of handlers
 */
export const handlers = [
  // mock slack reuqests
  rest.post('https://hooks.slack.com/services/*', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json({ ok: true }))
  }),
  rest.post('https://us4.api.mailchimp.com/3.0/lists/*', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json({ ok: true }))
  }),
  rest.get('https://production.shippingapis.com/*', (req, res, ctx) => {
    let mockPostalCodeResponse =
      '<?xml version="1.0" encoding="UTF-8"?>\n' +
      '<CityStateLookupResponse><ZipCode ID="0"><Zip5>90210</Zip5><City>BEVERLY HILLS</City><State>CA</State></ZipCode></CityStateLookupResponse>'
    let hasInvalidZip = false

    req.url.searchParams.forEach(value => {
      // mock failed zipcode lookup
      if (value.indexOf('99999') > -1) {
        hasInvalidZip = true
      }
    })

    if (hasInvalidZip) {
      mockPostalCodeResponse =
        '<?xml version="1.0" encoding="UTF-8"?>\n' +
        '<CityStateLookupResponse><ZipCode ID="0"><Error><Number>-2147219399</Number><Source>WebtoolsAMS;CityStateLookup</Source><Description>Invalid Zip Code.</Description><HelpFile/><HelpContext/></Error></ZipCode></CityStateLookupResponse>'
    }

    return res(ctx.status(200), ctx.xml(mockPostalCodeResponse))
  }),
]
