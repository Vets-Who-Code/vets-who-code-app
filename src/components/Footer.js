import React from 'react'
import Link from 'gatsby-link'

const Footer = () => (
  <div>
    <section className="footer-widgets pad-extra">
      <div className="container">
        <div className="row">
          <div className="col-md-4">
            <div className="widgets-content footer-widget-wrap">
              <h3 className="widgets-title">About #VetsWhoCode</h3>
              <p>
                FRAGO, doing business as #VetsWhoCode, is an exempt organization as described in
                Section 501(c)(3) of the Internal Revenue Code. Our EIN is 47-3555231.
              </p>
              <div className="footer-social">
                <a href="http://bit.ly/vetswhocode-fb" aria-label="Facebook">
                  <i className="fa fa-facebook-f" />
                </a>
                <a href="http://bit.ly/vets-who-code-twitter" aria-label="Twitter">
                  <i className="fa fa-twitter" />
                </a>
                <a href="http://bit.ly/2omsjdX" aria-label="Codepen">
                  <i className="fa fa-codepen" />
                </a>
                <a href="https://bit.ly/2sJ6W9f" aria-label="Instagram">
                  <i className="fa fa-instagram" />
                </a>
                <a href="https://bit.ly/2sFKi1u" aria-label="Linkedin">
                  <i className="fa fa-linkedin" />
                </a>
                <a href="http://bit.ly/vwc-github" aria-label="Github">
                  <i className="fa fa-github" />
                </a>
              </div>
            </div>
          </div>
          <div className="col-md-4">
            <div className="footer-widget-wrap">
              <div className="row">
                <div className="col-md-11 col-md-offset-1">
                  <h3 className="widgets-title">Site Map</h3>
                </div>
                <div className="col-md-5 col-md-offset-1">
                  <div className="widgets-content">
                    <ul className="widgets-list">
                      <li>
                        <Link to="/">Home</Link>
                      </li>
                      <li>
                        <Link to="/about">About</Link>
                      </li>
                      <li>
                        <Link to="/syllabus">Syllabus</Link>
                      </li>
                      <li>
                        <Link to="/mentor">Mentor</Link>
                      </li>
                    </ul>
                  </div>
                </div>
                <div className="col-md-6">
                  <div className="widgets-content">
                    <ul className="widgets-list">
                      <li>
                        <Link to="/apply">Apply</Link>
                      </li>
                      <li>
                        <Link to="/donate">Donate</Link>
                      </li>
                      <li>
                        <Link to="/contact">Contact Us</Link>
                      </li>
                      <li>
                        <a
                          href="https://medium.com/vets-who-code"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <span>Blog</span>
                        </a>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="col-md-4">
            <div className="widgets-content">
              <h3 className="widgets-title">Contact Us</h3>
              <div className="office map">
                <p>
                  <span>Hi, if you have any questions, please drop us a line.</span>
                  <br />
                  Email: hello@vetswhocode.io
                  <br />
                  Phone: (917) 960-3806
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <footer id="footer" className="section footer">
      <div className="container">
        <div className="row">
          <div className="col-md-6 col-sm-6">
            <p className="copyright">
              <span className="fa fa-copyright" /> #VetsWhoCode
            </p>
          </div>
          <div className="col-md-6 col-sm-6">
            <ul className="footer-menu">
              <li>
                Made with <span className="fa fa-heart red" /> By #VetsWhoCode.
              </li>
            </ul>
          </div>
        </div>
      </div>
    </footer>
  </div>
)

export default Footer
