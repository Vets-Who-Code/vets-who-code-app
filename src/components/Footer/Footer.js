import React from 'react'
import Link from 'gatsby-link'
import Icon from '../Icon'

function Footer() {
  return (
    <div>
      <section className="footer-widgets pad-extra">
        <div className="container">
          <div className="row">
            <div className="col-md-4">
              <div className="widgets-content footer-widget-wrap">
                <h3 className="widgets-title">About #VetsWhoCode</h3>
                <p>
                  FRAGO, doing business as #VetsWhoCode, is an exempt organization as described in
                  Section 501(c)(3) of the Internal Revenue Code. Our EIN is 47-3555231.
                </p>
                <div className="footer-social">
                  <a href="http://bit.ly/vetswhocode-facebook-link" aria-label="Facebook">
                    <Icon iconName="facebook-f" />
                  </a>
                  <a href="http://bit.ly/vets-who-code-twitter" aria-label="Twitter">
                    <Icon iconName="twitter" />
                  </a>
                  <a href="http://bit.ly/2omsjdX" aria-label="Codepen">
                    <Icon iconName="codepen" />
                  </a>
                  <a href="https://bit.ly/2sJ6W9f" aria-label="Instagram">
                    <Icon iconName="instagram" />
                  </a>
                  <a href="https://bit.ly/2sFKi1u" aria-label="Linkedin">
                    <Icon iconName="linkedin-in" />
                  </a>
                  <a href="http://bit.ly/vwc-github" aria-label="Github">
                    <Icon iconName="github" />
                  </a>
                </div>
              </div>
            </div>
            <div className="col-md-4">
              <div className="footer-widget-wrap">
                <div className="row">
                  <div className="col-md-11 col-md-offset-1">
                    <h3 className="widgets-title">Site Map</h3>
                  </div>
                  <div className="col-md-5 col-md-offset-1">
                    <div className="widgets-content">
                      <ul className="widgets-list">
                        <li>
                          <Link to="/">Home</Link>
                        </li>
                        <li>
                          <Link to="/about">About</Link>
                        </li>
                        <li>
                          <Link to="/syllabus">Syllabus</Link>
                        </li>
                        <li>
                          <Link to="/mentor">Mentor</Link>
                        </li>
                      </ul>
                    </div>
                  </div>
                  <div className="col-md-6">
                    <div className="widgets-content">
                      <ul className="widgets-list">
                        <li>
                          <Link to="/apply">Apply</Link>
                        </li>
                        <li>
                          <Link to="/donate">Donate</Link>
                        </li>
                        <li>
                          <Link to="/contact">Contact Us</Link>
                        </li>
                        <li>
                          <a
                            href="https://medium.com/vets-who-code"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            <span>Blog</span>
                          </a>
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className="col-md-4">
              <div className="widgets-content">
                <h3 className="widgets-title">Contact Us</h3>
                <div className="office map">
                  <p>
                    <span>Hi, if you have any questions, please drop us a line.</span>
                    <br />
                    Email: hello@vetswhocode.io
                    <br />
                    Phone: (917) 960-3806
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <footer id="footer" className="section footer">
        <div className="container">
          <div className="row">
            <div className="col-md-6 col-sm-6">
              <p className="copyright">
                <Icon iconName="copyright" />
                #VetsWhoCode
              </p>
            </div>
            <div className="col-md-6 col-sm-6">
              <ul className="footer-menu">
                <li>
                  Made with{' '}
                  <span className="fa fa-heart red">
                    <Icon iconName="heart" fill="red" />
                  </span>{' '}
                  By #VetsWhoCode.
                </li>
              </ul>
            </div>
          </div>
        </div>
      </footer>
    </div>
  )
}

export default Footer
