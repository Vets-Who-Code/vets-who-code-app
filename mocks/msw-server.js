import { rest } from 'msw'
import { setupServer } from 'msw/node'
import { handlers } from './handlers'

const mswServer = setupServer(...handlers)

export { rest, mswServer }
