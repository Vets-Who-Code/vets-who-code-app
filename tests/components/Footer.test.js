import { render } from '@testing-library/react'
import Footer from '@/components/Footer'
import { createMocks } from 'node-mocks-http'
import { mswServer, rest } from '@/mocks/msw-server'
import contactApiHandler from '@/api/contact'

describe('<Footer />', () => {
  test('should render correctly', () => {
    const { container } = render(<Footer />)
    expect(container.firstChild).toMatchSnapshot()
  })

  // Testing the Contact form in the footer
  test('should return 200 and success message', async () => {
    const body = {
      email: 'jody@mail.com',
      message: 'hello world',
    }

    const { req, res } = createMocks({
      method: 'POST',
      statusCode: 200,
      body: JSON.stringify(body),
    })

    await contactApiHandler(req, res)

    expect(res._getStatusCode()).toBe(200)
    expect(res._getData()).toBe('{"message":"SUCCESS"}')
  })

  test('should return 422 error when parameters missing', async () => {
    const body = {
      email: '',
      message: 'hello world',
    }

    const { req, res } = createMocks({
      method: 'POST',
      statusCode: 200,
      body: JSON.stringify(body),
    })

    await contactApiHandler(req, res)

    expect(res._getStatusCode()).toBe(422)
    expect(res._getData()).toBe('{"error":"Missing or incorrect required property"}')
  })

  test('should throw error when failing to complete post to slack api', async () => {
    // override default behaviour of handler use mock service worker to simulate a failed post to slack api
    mswServer.use(
      rest.post('https://hooks.slack.com/services/*', async (_, res, ctx) => {
        return res(ctx.status(500), ctx.json({}))
      })
    )

    const body = {
      email: 'jody@mail.com',
      message: 'hello world',
    }

    const { req, res } = createMocks({
      method: 'POST',
      statusCode: 200,
      body: JSON.stringify(body),
    })

    await contactApiHandler(req, res)

    expect(res._getStatusCode()).toBe(500)
    expect(res._getData()).toBe('{"message":"Failed post to #contact channel"}')
  })
})
