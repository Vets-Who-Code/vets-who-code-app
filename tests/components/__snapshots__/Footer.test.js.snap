// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
<div>
  <section
    class="footer-widgets pad-extra"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="widgets-content footer-widget-wrap"
          >
            <h3
              class="widgets-title"
            >
              About #VetsWhoCode
            </h3>
            <p>
              FRAGO, doing business as #VetsWhoCode, is an exempt organization as described in Section 501(c)(3) of the Internal Revenue Code. Our EIN is 47-3555231.
            </p>
            <div
              class="footer-social"
            >
              <a
                aria-label="Facebook"
                href="http://bit.ly/vetswhocode-facebook-link"
              >
                <i
                  class="fa"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="facebook-f"
                    data-prefix="fab"
                    focusable="false"
                    height="14"
                    role="img"
                    viewBox="0 0 320 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                      fill="#FFF"
                    />
                  </svg>
                </i>
              </a>
              <a
                aria-label="Twitter"
                href="http://bit.ly/vets-who-code-twitter"
              >
                <i
                  class="fa"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="twitter"
                    data-prefix="fab"
                    focusable="false"
                    height="14"
                    role="img"
                    viewBox="0 0 512 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                      fill="#FFF"
                    />
                  </svg>
                </i>
              </a>
              <a
                aria-label="Codepen"
                href="http://bit.ly/2omsjdX"
              >
                <i
                  class="fa"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="codepen"
                    data-prefix="fab"
                    focusable="false"
                    height="14"
                    role="img"
                    viewBox="0 0 512 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M502.285 159.704l-234-156c-7.987-4.915-16.511-4.96-24.571 0l-234 156C3.714 163.703 0 170.847 0 177.989v155.999c0 7.143 3.714 14.286 9.715 18.286l234 156.022c7.987 4.915 16.511 4.96 24.571 0l234-156.022c6-3.999 9.715-11.143 9.715-18.286V177.989c-.001-7.142-3.715-14.286-9.716-18.285zM278 63.131l172.286 114.858-76.857 51.429L278 165.703V63.131zm-44 0v102.572l-95.429 63.715-76.857-51.429L234 63.131zM44 219.132l55.143 36.857L44 292.846v-73.714zm190 229.715L61.714 333.989l76.857-51.429L234 346.275v102.572zm22-140.858l-77.715-52 77.715-52 77.715 52-77.715 52zm22 140.858V346.275l95.429-63.715 76.857 51.429L278 448.847zm190-156.001l-55.143-36.857L468 219.132v73.714z"
                      fill="#FFF"
                    />
                  </svg>
                </i>
              </a>
              <a
                aria-label="Instagram"
                href="https://bit.ly/2sJ6W9f"
              >
                <i
                  class="fa"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="instagram"
                    data-prefix="fab"
                    focusable="false"
                    height="14"
                    role="img"
                    viewBox="0 0 448 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                      fill="#FFF"
                    />
                  </svg>
                </i>
              </a>
              <a
                aria-label="Linkedin"
                href="https://bit.ly/2sFKi1u"
              >
                <i
                  class="fa"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="linkedin-in"
                    data-prefix="fab"
                    focusable="false"
                    height="14"
                    role="img"
                    viewBox="0 0 448 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                      fill="#FFF"
                    />
                  </svg>
                </i>
              </a>
              <a
                aria-label="Github"
                href="http://bit.ly/vwc-github"
              >
                <i
                  class="fa"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="github"
                    data-prefix="fab"
                    focusable="false"
                    height="14"
                    role="img"
                    viewBox="0 0 496 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                      fill="#FFF"
                    />
                  </svg>
                </i>
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="footer-widget-wrap"
          >
            <div
              class="row"
            >
              <div
                class="col-md-11 col-md-offset-1"
              >
                <h3
                  class="widgets-title"
                >
                  Site Map
                </h3>
              </div>
              <div
                class="col-md-5 col-md-offset-1"
              >
                <div
                  class="widgets-content"
                >
                  <ul
                    class="widgets-list"
                  >
                    <li>
                      <a
                        aria-current="page"
                        class=""
                        href="/"
                      >
                        Home
                      </a>
                    </li>
                    <li>
                      <a
                        href="/about"
                      >
                        About
                      </a>
                    </li>
                    <li>
                      <a
                        href="/syllabus"
                      >
                        Syllabus
                      </a>
                    </li>
                    <li>
                      <a
                        href="/mentor"
                      >
                        Mentor
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
              <div
                class="col-md-6"
              >
                <div
                  class="widgets-content"
                >
                  <ul
                    class="widgets-list"
                  >
                    <li>
                      <a
                        href="/apply"
                      >
                        Apply
                      </a>
                    </li>
                    <li>
                      <a
                        href="/donate"
                      >
                        Donate
                      </a>
                    </li>
                    <li>
                      <a
                        href="/contact"
                      >
                        Contact Us
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://medium.com/vets-who-code"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        <span>
                          Blog
                        </span>
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="widgets-content"
          >
            <h3
              class="widgets-title"
            >
              Contact Us
            </h3>
            <div
              class="office map"
            >
              <p>
                <span>
                  Hi, if you have any questions, please drop us a line.
                </span>
                <br />
                Email: hello@vetswhocode.io
                <br />
                Phone: (917) 960-3806
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <footer
    class="section footer"
    id="footer"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6 col-sm-6"
        >
          <p
            class="copyright"
          >
            <i
              class="fa"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="copyright"
                data-prefix="fab"
                focusable="false"
                height="14"
                role="img"
                viewBox="0 0 512 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 448c-110.532 0-200-89.451-200-200 0-110.531 89.451-200 200-200 110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200zm107.351-101.064c-9.614 9.712-45.53 41.396-104.065 41.396-82.43 0-140.484-61.425-140.484-141.567 0-79.152 60.275-139.401 139.762-139.401 55.531 0 88.738 26.62 97.593 34.779a11.965 11.965 0 0 1 1.936 15.322l-18.155 28.113c-3.841 5.95-11.966 7.282-17.499 2.921-8.595-6.776-31.814-22.538-61.708-22.538-48.303 0-77.916 35.33-77.916 80.082 0 41.589 26.888 83.692 78.277 83.692 32.657 0 56.843-19.039 65.726-27.225 5.27-4.857 13.596-4.039 17.82 1.738l19.865 27.17a11.947 11.947 0 0 1-1.152 15.518z"
                  fill="#FFF"
                />
              </svg>
            </i>
            #VetsWhoCode
          </p>
        </div>
        <div
          class="col-md-6 col-sm-6"
        >
          <ul
            class="footer-menu"
          >
            <li>
              Made with
               
              <span
                class="fa fa-heart red"
              >
                <i
                  class="fa"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="heart"
                    data-prefix="fab"
                    focusable="false"
                    height="14"
                    role="img"
                    viewBox="0 0 512 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                      fill="red"
                    />
                  </svg>
                </i>
              </span>
               
              By #VetsWhoCode.
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
